tools/syz-declextract/run.go

```diff
@@ -72,15 +112,61 @@ func worker(outputs chan output, files chan string, binary, compilationDatabase
 
 		cmd := exec.Command(binary, "-p", compilationDatabase, file)
 		stdout, err := cmd.Output()
-		var stderr []byte
+		var stderr string
 		if err != nil {
 			var error *exec.ExitError
 			if errors.As(err, &error) {
-				stderr = error.Stderr
+				stderr = string(error.Stderr)
 			} else {
-				stderr = []byte(err.Error())
+				stderr = err.Error()
+			}
+		}
+		outputs <- output{string(stdout), stderr}
+	}
+}
+
+func renameSyscall(kernelDir string) func(string) string {
+	var rename = make(map[string]string)
+	var isValidSyscall = make(map[string]bool)
+	for _, arch := range targets.List[targets.Linux] {
+		filepath.Walk(kernelDir+arch.KernelHeaderArch, func(path string, info fs.FileInfo, err error) error {
+			if strings.HasSuffix(path, ".tbl") {
+				fi, err := os.Lstat(path)
+				if err != nil {
+					tool.Fail(err)
+				}
+				if fi.Mode()&fs.ModeSymlink != 0 { // Don't follow symlinks.
+					return nil
+				}
+				f, err := os.Open(path)
+				if err != nil {
+					tool.Fail(err)
+				}
+				s := bufio.NewScanner(f)
+				for s.Scan() {
+					fields := strings.Fields(s.Text())
+					if len(fields) < 4 || fields[0] == "#" || fields[3] == "-" || strings.HasPrefix(fields[3], "compat") ||
+						fields[3][4:] == "ni_syscall" {
+						continue
+					}
+					rename[fields[3][4:]] = fields[2]
+					isValidSyscall[fields[2]] = true
+				}
+			}
+			return nil
+		})
+	}
+
+	return func(desc string) string {
+		if desc != "" {
+			toReplace := strings.Split(desc, "$")[0]
+			if isValidSyscall[toReplace] { // Do not replace if the syscall is valid.
+				return desc
+			} else if rename[toReplace] != "" {
+				return strings.Replace(desc, toReplace, rename[toReplace], 1)
```

Comment:
Can't the same syscall be renamed to 2 or more names in different arches?

Reply:
It can. All valid names should be present now!